src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs

```diff
@@ -52,9 +52,18 @@ private CompletionProvider GetCompletionProvider(CompletionItem item)
             this.StopModelComputation();
 
             CompletionChange completionChange;
-            using (var transaction = new CaretPreservingEditTransaction(
+            using (var transaction = CaretPreservingEditTransaction.TryCreate(
                 EditorFeaturesResources.IntelliSense, TextView, _undoHistoryRegistry, _editorOperationsFactoryService))
             {
+                if (transaction == null)
+                {
+                    // This text buffer has no undo history and has probably been unmapped.
```

Comment:
I'm not sure I understand this comment. We always should be getting the undo history from the surface buffer precisely because that's where it lives. So how does mapping have anything to do with it?